crml/sylo/src/directory/rpc/src/lib.rs

```diff
@@ -49,19 +50,26 @@ impl From<Error> for i64 {
 	}
 }
 
-impl<C, Block, Balance, AccountId> SyloDirectoryApi<Balance, AccountId> for SyloDirectory<C, Block>
+impl<C, Block, AccountId> SyloDirectoryApi<AccountId> for SyloDirectory<C, Block>
 where
 	Block: BlockT,
 	C: Send + Sync + 'static + ProvideRuntimeApi<Block> + HeaderBackend<Block>,
-	C::Api: SyloDirectoryRuntimeApi<Block, Balance, AccountId>,
-	Balance: Codec + BaseArithmetic,
+	C::Api: SyloDirectoryRuntimeApi<Block, AccountId>,
 	AccountId: Codec,
 {
-	fn scan(&self, point: Balance) -> Result<AccountId> {
+	fn scan(&self, point: U256) -> Result<AccountId> {
 		let api = self.client.runtime_api();
 		let best = self.client.info().best_hash;
 		let at = BlockId::hash(best);
 
+		// Accepts u256 but should only work on u128
+		if point > u128::MAX.into() {
+			return Err(RpcError {
+				code: ErrorCode:: ServerError(Error::Runtime.into()),
+				message: "Value too large.".into(),
+				data: Some(format!("Expected a 128bit integer but received {:?} instead", point).into()), 
```

Comment:
I feel like you would only understand this error if you understood the internals of the problem here. As a fresh developer I don't know if this would help me. But I can't think of a succinct error to put here instead ...

Reply:
Yeah I wasn't quite sure what to put there.